exercises/simple-linked-list/linked_list.exs

```diff
@@ -0,0 +1,73 @@
+defmodule LinkedList do
+  @opaque t :: any()
+
+  @doc """
+  Construct a new LinkedList
+  """
+  @spec new() :: t
+  def new() do
+  end
+
+  @doc """
+  Push an item onto a LinkedList
+  """
+  @spec push(t, any()) :: t
+  def push(list, elem) do
+  end
+
+  @doc """
+  Calculate the length of a LinkedList
+  """
+  @spec length(t) :: non_neg_integer()
+  def length(list) do
+  end
+
+  @doc """
+  Determine if a LinkedList is empty
+  """
+  @spec empty?(t) :: boolean()
+  def empty?(list) do
+  end
+
+  @doc """
+  Get the value of a head of the LinkedList
+  """
+  @spec peek(t) :: {:ok, any()} | {:error, :empty_list}
+  def peek(list) do
+  end
+
+  @doc """
+  Get tail of a LinkedList
+  """
+  @spec tail(t) :: {:ok, t} | {:error, :empty_list}
+  def tail(list) do
+  end
+
+  @doc """
+  Remove the head from a LinkedList
```

Comment:
"Remove the head from a linked list, and return the value and the tail of the linked list"???